did/plugins/zammad.py

```diff
@@ -0,0 +1,121 @@
+# coding: utf-8
+"""
+Zammad stats such as updated tickets
+
+Config example::
+
+    [zammad]
+    type = zammad
+    url = https://zammad.example.com/api/v1/
+    token = <authentication-token>
+
+"""
+
+import json
+import urllib.request, urllib.error, urllib.parse
+
+from did.utils import log, pretty, listed
+from did.base import Config, ReportError
+from did.stats import Stats, StatsGroup
+
+# Identifier padding
+PADDING = 3
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#  Investigator
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+class Zammad(object):
+    """ Zammad Investigator """
+
+    def __init__(self, url, token):
+        """ Initialize url and headers """
+        self.url = url.rstrip("/")
+        if token is not None:
+            self.headers = {'Authorization': 'Token token={0}'.format(token)}
+        else:
+            self.headers = {}
+
+        self.token = token
+
+    def search(self, query):
+        """ Perform Zammad query """
+        url = self.url + "/" + query
+        log.debug("Zammad query: {0}".format(url))
+        try:
+            request = urllib.request.Request(url, headers=self.headers)
+            response = urllib.request.urlopen(request)
+            log.debug("Response headers:\n{0}".format(
+                str(response.info()).strip()))
+        except urllib.error.URLError as error:
+            log.debug(error)
+            raise ReportError(
+                "Zammad search on {0} failed.".format(self.url))
+        result = json.loads(response.read())["assets"]["Ticket"]
```

Comment:
```suggestion
        try:
            result = json.loads(response.read())["assets"]["Ticket"]
        except KeyError:
            result =  dict() 
```
Problem is that if  API changes this will happily return empty results.

Reply:
Hi @come-nc. Another option is first do `result = json.loads(response.read())["assets"]` and after that try/except `result = result['Ticket']`

Reply:
Thank you, it works.